elasticsearchIndexing/process-changed-mods.py

```diff
@@ -23,186 +23,223 @@
-            LOGGER.info('Index {} created with message:\n{}'.format(index.value, create_result))
-
-    logging.getLogger('elasticsearch').setLevel(logging.ERROR)
-
-    LOGGER.info('Running cache files backup')
-    backup_cache_files(delete_cache_path)
-    backup_cache_files(changes_cache_path)
-    os.unlink(lock_file)
-
-    if delete_cache:
-        for module in delete_cache:
+class ScriptConfig(BaseScriptConfig):
+
+    def __init__(self):
+        help = 'Process changed modules in a git repo'
+        args: t.List[Arg] = [
+            {
+                'flag': '--config-path',
+                'help': 'Set path to config file',
+                'type': str,
+                'default': os.environ['YANGCATALOG_CONFIG_PATH']
+            },
+        ]
+        super().__init__(help, args, None if __name__ == '__main__' else [])
+        
+        
+class ProcessChangedMods:
+    def __init__(self, script_config: BaseScriptConfig):
+        self.args = script_config.args
+        self.config = create_config(self.args.config_path)
+        self.log_directory = self.config.get('Directory-Section', 'logs')
+        self.yang_models = self.config.get('Directory-Section', 'yang-models-dir')
+        self.changes_cache_path = self.config.get('Directory-Section', 'changes-cache')
+        self.failed_changes_cache_path = self.config.get('Directory-Section', 'changes-cache-failed')
+        self.delete_cache_path = self.config.get('Directory-Section', 'delete-cache')
+        self.lock_file = self.config.get('Directory-Section', 'lock')
+        self.lock_file_cron = self.config.get('Directory-Section', 'lock-cron')
+        self.json_ytree = self.config.get('Directory-Section', 'json-ytree')
+        self.save_file_dir = self.config.get('Directory-Section', 'save-file-dir')
+
+        self.logger = log.get_logger(
+            'process_changed_mods',
+            os.path.join(self.log_directory, 'process-changed-mods.log'),
+        )
+    
+    def start_processing_changed_mods(self):
+        self.logger.info('Starting process-changed-mods.py script')
+
+        if os.path.exists(self.lock_file) or os.path.exists(self.lock_file_cron):
+            # we can exist since this is run by cronjob every 3 minutes of every day
+            self.logger.warning('Temporary lock file used by something else. Exiting script !!!')
+            sys.exit()
+        self._create_lock_files()
+
+        self.changes_cache = self._load_changes_cache(self.changes_cache_path)
+        self.delete_cache = self._load_delete_cache(self.delete_cache_path)
+        
+        if not self.changes_cache and not self.delete_cache:
+            self.logger.info('No new modules are added or removed. Exiting script!!!')
+            os.unlink(self.lock_file)
+            os.unlink(self.lock_file_cron)
+            sys.exit()
+
+        self.logger.info('Pulling latest YangModels/yang repository')
+        repoutil.pull(self.yang_models)
```

Comment:
I also have a question about pulling yang_models every time the script executes, what's the point of pulling models? I don't see where these models are used then, maybe I'm missing something.

Reply:
Hmm, I'm not 100% sure either. Looks like it was introduced in [this commit](https://github.com/YangCatalog/backend/commit/7236d4be063fcca99a230e6d1cddf3fca70a0c96#diff-edbe1631ff443625e1373391fe15a0cde434a4905188667d30d94975b9a448aa). @SlavomirMazurPantheon do you know if this is important for something?

Reply:
@SlavomirMazurPantheon it takes a lot of time to finish as well, so if it's not important I'll better remove it

Reply:
I am not sure why I put this here - it can be removed, because this script runs every 3 minutes while the yangmodels repo is updated 1-2 a day so it is not necessary to check changes every 3 minutes.
In fact, maybe this was the reason, that I wanted this local cloned repo to be as actual as possible.

However, question is why it is taking so much time when it should perform just `git pull`?

Reply:
I just wanted to check how much time it takes to make a pull request here but couldn't do it because I have a broken ```config``` file in ```/var/yang/nonietf/yangmodels/yang/.git```, so @richardzilincikPantheon @SlavomirMazurPantheon could anyone please share his ```config``` file with me and I will check it then?
I didn't notice an error earlier, so I believe this part doesn't really consume so much time because I just commented it earlier to check if logging works, as I saw that the script wasn't passing this ```pull``` request for a long time